### PR TITLE
CFE-2977:  A bootstrap_id.dat file is now generated on every bootstrap

### PR DESCRIPTION
**Pull request by olehermanse**

### CODE REVIEW

**Comment by olehermanse** [comment]
@cf-bottom jenkins

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/2025/)

https://ci.cfengine.com/job/pr-pipeline/2025/

File: cf-agent/cf-agent.c [hunk 1]
```diff
@@ -324,6 +324,74 @@ int main(int argc, char *argv[])
 /* Level 1                                                         */
 /*******************************************************************/
 
+static void BootstrapConfig(GenericAgentConfig *config, const char *argument)
```
Comment (vpodzime) [inline cf-agent/cf-agent.c]: Why not just `Bootstrap()`?

File: libpromises/bootstrap.c [hunk 2]
```diff
@@ -353,3 +356,31 @@ bool MasterfileExists(const char *masterdir)
...
+    FILE *file = fopen(filename, "w");
+    if (file == NULL)
+    {
+        Log(LOG_LEVEL_ERR, "Unable to write bootstrap id file '%s' (fopen: %s)", filename, GetErrorStr());
+        free(filename);
+        return false;
+    }
+    CryptoInitialize();
+    unsigned char buf[8];
+    RAND_bytes(buf, 8);
```
Comment (vpodzime) [inline libpromises/bootstrap.c]: Any reason for not using at least 128 bits? I know `2**64` is a big number, but some people are paranoid about these things.

File: cf-agent/cf-agent.c (hunk 1 above)
Comment (olehermanse) [reply cf-agent/cf-agent.c]: Bootstrapping is actually done in `GenericAgentDiscoverContext()`, this code just sets up the config correctly for later.

Comment (vpodzime) [reply cf-agent/cf-agent.c]: Yeah, I saw that in the second commit. What about `ConfigureBootstrap()` then?

Comment (olehermanse) [reply cf-agent/cf-agent.c]: Fair enough.

File: libpromises/bootstrap.c (hunk 2 above)
Comment (olehermanse) [reply libpromises/bootstrap.c]: Increased to 240 bits, which is divisible by 6, meaning there won't be base64 padding.

**Comment by olehermanse** [comment]
@cf-bottom jenkins

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/2059/)

https://ci.cfengine.com/job/pr-pipeline/2059/

**Comment by olehermanse** [comment]
I think this can be merged even without the sys variable.

**Pull request merged**